src/repl.coffee

```diff
@@ -26,7 +26,7 @@ replDefaults =
       ast = new Block [
         new Assign (new Value new Literal '_'), ast, '='
       ]
-      js = ast.compile bare: yes, locals: Object.keys(context)
+      js = ast.compile bare: yes, locals: Object.keys(context), referencedVars: []
```

Comment:
Better yet: tokenize `input`, filter out all referenced variables then pass the tokens to `CoffeeScript.nodes`.